src/resultset/resultset.c

```diff
@@ -145,28 +146,33 @@ void _aggregateResultSet(RedisModuleCtx* ctx, ResultSet* set) {
     FreeGroupCache(set->groups);
 }
 
-/* TODO: Drop heap, use some sort algo. */
-Record** _sortResultSet(const ResultSet *set, const Vector* records) {
-    size_t len = Vector_Size(records);
+static int _record_compare(Record *a, Record *b, const ResultSet *set) {
+    int idx = 0;
+    int rel;
+    for(int i = 0; i < set->header->orderby_len; i++) {
+      idx = set->header->orderBys[i];
+      rel = SIValue_Compare(a->values[idx], b->values[idx]);
+      rel *= set->direction; // flip value for descending order
+      if (rel < 0) {
+          return 0;
+      } else if (rel > 0) {
+          return 1;
+      }
+      // If rel == 0, proceed to the next ORDER BY element
+    }
+    return 0;
+}
 
-    Record **arrRecords = malloc(sizeof(Record*) * len);
-    heap_t *heap = heap_new(_heap_elem_compare, set);
+/* `set` is an actual variable in the caller function. Using it in a
+ * macro like this is rather ugly, but the macro passed to QSORT must
+ * accept only 2 arguments. */
+#define RECORD_SORT(a, b) _record_compare((*a), (*b), set)
```

Comment:
It's definitely a bit fragile, but I prefer it over introducing a global variable or wrapping records in a struct.

I feel like it's adequate for the moment, though I agree that it would be nice to replace it with something more straightforward later.